projects/templates/src/lib/components/po-page-dynamic-detail/po-page-dynamic-detail.component.spec.ts

```diff
@@ -158,121 +166,286 @@ describe('PoPageDynamicDetailComponent:', () => {
+            { property: 'filter3' }
+          ]);
+          expect(component.breadcrumb).toEqual({
+            items: [
+              { label: 'Test' },
+              { label: 'Test2' }
+            ]
+          });
+
+          component['subscriptions'] = null;
+          component.ngOnDestroy();
+        }));
+
+      it('should configure properties based on the return of onload route', fakeAsync(() => {
+
+        component.autoRouter = false;
+        component.actions = <any>{};
+        component.breadcrumb = <any>{};
+        component.fields = [];
+        component.title = '';
 
-    it('ngOnInit: should call `loadMetadata` with `id` and set `serviceApi` if `activatedRoute.snapshot.data.serviceApi` is truthy',
-      () => {
-        const id = 1;
         const activatedRoute: any = {
           snapshot: {
             data: {
-              serviceApi: 'localhost:4300/api/people'
+              serviceApi: 'localhost:4300/api/people',
+              serviceMetadataApi: 'localhost:4300/api/people/metadata',
+              serviceLoadApi: 'localhost:4300/api/people/metadata'
             },
-            params: { id }
+            params: { id: 1 }
           }
         };
 
-        component.serviceApi = undefined;
+        const metadata = {
+          breadcrumb : {
+            items: [
+              { label: 'Home' },
+              { label: 'Hiring processes' }
+            ]
+          },
+          title: 'Original Title'
+        };
 
-        spyOn(component, <any> 'loadMetadata');
-        spyOn(component['poPageDynamicService'], <any> 'configServiceApi');
+        const custom = {title: 'New Title'};
+
+        spyOn(component, <any>'loadData').and.returnValue(of({}));
+        spyOn(component['poPageDynamicService'], 'getMetadata').and.returnValue(of(metadata));
+        spyOn(<any>component['poPageCustomizationService'], 'createObservable').and.returnValue(of(custom));
 
         component['activatedRoute'] = activatedRoute;
 
         component.ngOnInit();
 
-        expect(component.serviceApi).toEqual(activatedRoute.snapshot.data.serviceApi);
-        expect(component['loadMetadata']).toHaveBeenCalledWith(id);
-        expect(component['poPageDynamicService'].configServiceApi).toHaveBeenCalledWith({ endpoint: component.serviceApi });
-      });
+        tick();
 
-    it('loadMetadata: should call `getMetadata` and set properties', fakeAsync(() => {
-      const id = 1;
-      const response = {
-        autoRouter: false,
-        actions: undefined,
-        breadcrumb: undefined,
-        fields: [],
-        title: 'Title'
-      };
+        expect(component.title).toBe('New Title');
+        expect(component.breadcrumb).toEqual({
+            items: [
+              { label: 'Home' },
+              { label: 'Hiring processes' }
+            ]
+          });
+        }));
+    });
+
+    describe('loadDataFromAPI', () => {
+      it('should call `getMetadata` and set properties', fakeAsync(() => {
```

Comment:
**Sugestão:**
colocar no título do teste, algo que referencie o comportamento relacionado ao merge com propriedades vazias.
Por exemplo: deve manter os dados do getMetadata se loadOptionsOnInitialize for vazio.

Reply:
Feito